### PR TITLE
feat(leaderboard): retrieve and show leaderboard on client

### PR DESCRIPTION
**Pull request by silimarius**
## Description

This PR integrates leaderboard functionality between backend and frontend

## Changes Made

- tRPC endpoint for fetching leaderboard data
- Infinite query to fetch leaderboard data on scroll and populate the table
- Script to populate users db and leaderboard when developing

### CODE REVIEW

File: src/constants/fake-data/leaderboard.ts [hunk 1]
```diff
@@ -6,284 +6,222 @@ export const leaderboardData: LeaderboardData[] = [
   {
     rank: 1,
     score: 10000,
-    address: "1BoatSLRHtKNngkdXEeobR76b53LETtpyT",
-    avatar: "🙂",
     username: "AliceInWonderland",
-    gamesPlayed: 50,
-    payout: 30,
+    matchesPlayed: ["a", "b"],
```
Comment (ramidecodes) [inline src/constants/fake-data/leaderboard.ts:10]: Should this string comply with the UUID type?

File: src/pages/leaderboard.tsx [hunk 2]
```diff
@@ -1,70 +1,81 @@
...
   const { data: sessionData } = useSession();
   const isAuthenticated = isValidSession(sessionData);
+
   const isGamePlayed = true;
 
-  const loadMoreData = useCallback(() => {
-    const startIndex: number = (currentPage - 1) * itemsPerPage;
-    const endIndex: number = startIndex + itemsPerPage;
-    const newData: LeaderboardData[] = leaderboardData.slice(0, endIndex);
+  const getRankedUsers = api.user.getRankedUsers.useInfiniteQuery(
```
Comment (ramidecodes) [inline src/pages/leaderboard.tsx:21]: Shouldn't this be imported from db or some other service?

File: src/utils/leaderboard.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+export const calcPayout = (rank: number) => {
```
Comment (ramidecodes) [inline src/utils/leaderboard.ts:1]: This more likely will be disabled until further notice.

**Review by ramidecodes — APPROVED** [review]

File: src/constants/fake-data/leaderboard.ts (hunk 1 above)
Comment (silimarius) [reply src/constants/fake-data/leaderboard.ts:10]: This file will be deleted so it doesn't matter

File: src/pages/leaderboard.tsx (hunk 2 above)
Comment (silimarius) [reply src/pages/leaderboard.tsx:21]: I doubt it would be a good idea to call the DB from the client

File: src/utils/leaderboard.ts (hunk 3 above)
Comment (silimarius) [reply src/utils/leaderboard.ts:1]: It's in the designs so I put it there. Should I just yeet the column out?

**Pull request merged**